src/polyswarm_api/api.py

```diff
@@ -823,7 +828,51 @@ def event_list(self, **kwargs):
         logger.info('List events')
         return resources.Events.list(self, **kwargs).result()
 
-    def __repr__(self):
-        clsname = '{0.__module__}.{0.__name__}'.format(self.__class__)
-        attrs = 'uri={0.uri!r}, community={0.community!r}, timeout={0.timeout!r}'.format(self)
-        return '<{}({}) at 0x{:x}>'.format(clsname, attrs, id(self))
+    def report_create(self, **kwargs):
+        return resources.ReportTask.create(self, **kwargs).result()
+
+    def report_get(self, **kwargs):
+        return resources.ReportTask.get(self, **kwargs).result()
+
+    def report_download(self, report_id, folder):
+        report = resources.ReportTask.get(self, id=report_id).result()
+        result = report.download_report(folder=folder).result()
+        result.handle.close()
+        return result
+
+    def report_template_logo_download(self, template_id, folder):
+        report = resources.ReportTemplate.get(self, id=template_id).result()
+        result = report.download_logo(folder).result()
+        result.handle.close()
+        return result
+
+    def report_template_logo_delete(self, template_id):
+        report = resources.ReportTemplate.get(self, id=template_id).result()
+        result = report.delete_logo().result()
+        return result
+
+    def report_template_logo_upload(self, template_id, logo_file, content_tpe):
+        report = resources.ReportTemplate.get(self, id=template_id).result()
+        result = report.upload_logo(logo_file, content_tpe).result()
+        return result
+
+    def report_template_create(self, **kwargs):
+        return resources.ReportTemplate.create(self, **kwargs).result()
+
+    def report_template_update(self, template_id, **kwargs):
```

Comment:
I believe this method can encompass the `report_template_default()` by using a variable, so that we don't have another method just for setting the default really. Both of them just use the `ReportTemplate.update()` method, but this one is more generic. The signature here can be
```
def report_template_update(self, template_id, is_default=False, **kwargs):
```
and it can pass these parameters along to the update.

Reply:
Yes originally was like that, I just wanted a shortcut because defaulting a template is more than editing a single template, it is also un-defaulting the previous default template (if there was one). API-wise yes it's the same because the API takes care of that.

Anyway, removing ...

Reply:
My argument for this is kind of philosophical, but I can quote the zen of python 

```
There should be one-- and preferably only one --obvious way to do it.
```
and for me we should have only one method that does update in this case.

Reply:
Removed.